Pull Request

ethanwright3 opened the pull request:
I think everything is in as it should. Here is the pull request you asked for.

neilhatfield committed:
Initial Push

This is the initial push moving code from the old repo into the new one.

ethanwright3 committed:
push connection

I've connected the dev 1 to my own files successfully it appears.

ethanwright3 committed:
Fix App

Think I somehow screwed up first request sending in sample app instead of right one. But here is the right one working as it should.

neilhatfield committed:
Fix Files

Fixed DESCRIPTION and README.md

neilhatfield committed:
Clean up DESCRIPTION

Cleaned the full title in DESCRIPTION and removed index.htm

neilhatfield reviewed (changes requested):
Be sure that you pull the commits (see be581ff) BEFORE you begin making changes.

I'm going include the comments that I emailed you.

1. Please put the App's title as the h1 header on the Overview.
2. Add at least one sentence about the app just below this header and before the instructions.
3. Please add end of sentence punctuation to the instructions (and everywhere else).
4. The i button currently doesn't do anything when clicked? Is this supposed to be that way? If not, please fix. If yes, remove.
5. Place "Prerequisites" as the h2 header on the Prerequisites page. Make everything else, part of a paragraph.
    + I would add what MSOS is to the Prerequisites page (at bare minimum, what this initialism stands for).
6. For the intext hyperlink, please add class = "bodylinks".
7. Where the References page? Add and be sure to include the Machine Learning Cheatsheet you're linking to.
8. Use Title Case for the headers (e.g., "Explore Testing Percentage")
9. Please do not use h3 for the titles of boxes
10. I'll be honest, I'm not digging the layout, particularly the boxes for the data table and variable information.
    + Let's use a tab set panel that has either two or three tabs: the plots (either together or separate), and then the data table. That way, the table is there, a user can see it if they want, but it is unobtrusive.
    + I'd rather the variable information not be hide-able at all (i.e., it's always showing)
    + Why are some attributes bold face?
    + Also, the Iris variables aren't enclosed in a tags$ul; you just have a bunch of tags$li
11. User Controls
    + Capitalize N in "Number of repetitions..."
    + 3000 seems extremely excessive and is rather slow; consider dropping and just using 100
    + Switch the position of the slider and button (slider first, then button)
    + Change the label of the slider to "Proportion of data in training set"
    +I found myself wanting a "Reset" button so that I could stick with the same dataset and variable but just start over.
12. Plots
    + Change the second plot's title to "Mean Accuracy"
    + Update vertical labels to read "Mean _[current labels]_ " for the second plot
    + The upper plot has a fixed horizontal axis, which I like. The lower plot does not, which I don't like. Use the fixed axis for both plots.
    + I'm okay with the shifting/updating vertical axis for the first plot, but not for the second plot (Mean). See my email for the photo of the consequences.
    + For the first plot, use alpha = 0.5 as a starting point in the geom_point call. This will make the points 50% transparent. The more points in a certain location, the darker it will become.
    + Spell out Percentage, don't use %
    + Use sentence case for axis and input labels; e.g., only the first word is capitalized and proper nouns.
13. I've reformatted the code in your app. You'll need to work from this reformatting.
14. See Issue #2 
15. Please be explicit in your code and format your code for readability. For example, you have  the following:

```
column(2,
                   selectInput(inputId = 'theDataSet', label = 'Dataset', choices = list('Heart Disease', 'Palmer Penguins', 'Marketing','Iris'), selected = 'Heart Disease'),
                   selectInput(inputId = 'theVariable', label = 'Variable to predict', choices = list('Species', 'Sepal.Length', 'Sepal.Width', 'Petal.Length', 'Petal.Width'), selected = 'Species'),
                   ),
```
The following is what you should have:

```
column(
  width = 2,
  selectInput(
    inputId = 'theDataSet',
    label = 'Dataset',
    choices = list(
      'Heart Disease',
      'Palmer Penguins',
      'Marketing',
      'Iris'
    ),
    selected = 'Heart Disease'
  ),
 ```

16. Remove any extra white space (particularly empty lines), extraneous comments, and commented code (e.g., debugging statements)
17. Rather than having the many `if-else if` statements (see lines 221-324), try to condense them. For example, you could use the following: 

```
if (input$theVariable %in% c("Species", "species", "island", "sex", "target") {
  yLabel <- "Percentage of correct predictions"
} else {
  yLabel <- "MSOS of the error (smaller is better)"
}
```

18. Read all of the data files in at the top of your app (see where I placed `data(iris)`. By reading the files in down in the server code, you're app will have to load the data for every single user whereas above, it can share the loaded data among several users
19. I see how you made the ordered listed (via direct HTML tags). I've reformatted them so that they are 1) formatted correctly, and 2) using proper tags.
20. Please move any code you're wanting to keep in the app for future development to the section marked "Old/Future (?) Code" located at the bottom of the app.

There are additional things that can be down to clean up your code. However, please focus on getting the above done ASAP.

neilhatfield committed:
Update app.R

Updated App file

neilhatfield commented:
Also, I commented out some packages that I don't believe that you're actually using in this version of the app.

ethanwright3 committed:
Before meeting

updates to professor pearl and hatfield

ethanwright3 committed:
Hatfield and Pearl fixes

Here is all of the updates asked by Professor Pearl and Hatfield. The only exceptions being reset button, the references, and haven't cleaned packages.

ethanwright3 committed:
label fix

label fix

ethanwright3 committed:
Commit taking in edits

This takes in the edits from professor hatfield and hopefully next one will do my edits on his edits smoothly

neilhatfield committed:
Code Polishing

Neil's edits

neilhatfield reviewed (approved):
Approving the changes that I uploaded
